# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/nsHTTPSOnlyUtils.h
# Commit: 9d14d5ae3863
# Full Hash: 9d14d5ae386341911507881b91c4e709795a9395
# Author: lyavor <lyavor@mozilla.com>
# Date: 2021-04-26 15:46:32
# Regressor Bug: 1683015
# File Overlap Count: 1
# Description:
#   Bug 1683015 HTTPS-Only Mode Alert appears on site which supports https. r=ckerschb,JulianWels
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D112702
# ==============================================================================

diff -r 87457b540267 -r 9d14d5ae3863 dom/security/nsHTTPSOnlyUtils.h
--- a/dom/security/nsHTTPSOnlyUtils.h	Mon Apr 26 11:37:49 2021 +0000
+++ b/dom/security/nsHTTPSOnlyUtils.h	Mon Apr 26 11:45:20 2021 +0000
@@ -217,6 +217,14 @@
   ~TestHTTPAnswerRunnable() = default;
 
  private:
+  /**
+   * Checks whether the HTTP background request results in a redirect
+   * to the same upgraded top-level HTTPS URL
+   * @param  aChannel a nsIHttpChannel object
+   * @return  true if the backgroundchannel is redirected
+   */
+  static bool IsBackgroundRequestRedirected(nsIHttpChannel* aChannel);
+
   RefPtr<nsIURI> mURI;
   // We're keeping a reference to DocumentLoadListener instead of a specific
   // channel, because the current top-level channel can change (for example